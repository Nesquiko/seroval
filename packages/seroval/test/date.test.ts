import { describe, it, expect } from 'vitest';
import {
  deserialize,
  fromJSON,
  serialize,
  serializeAsync,
  toJSONAsync,
  toJSON,
} from '../src';

describe('Date', () => {
  describe('serialize', () => {
    it('supports Date', () => {
      const example = new Date('2023-03-14T11:16:24.879Z');
      const result = serialize(example);
      expect(result).toMatchSnapshot();
      const back = deserialize<Date>(result);
      expect(back).toBeInstanceOf(Date);
      expect(back.toISOString()).toBe(example.toISOString());
    });
  });
  describe('serializeAsync', () => {
    it('supports Date', async () => {
      const example = new Date('2023-03-14T11:16:24.879Z');
      const result = await serializeAsync(Promise.resolve(example));
      expect(result).toMatchSnapshot();
      const back = await deserialize<Promise<Date>>(result);
      expect(back).toBeInstanceOf(Date);
      expect(back.toISOString()).toBe(example.toISOString());
    });
  });
  describe('toJSON', () => {
    it('supports Date', () => {
      const example = new Date('2023-03-14T11:16:24.879Z');
      const result = toJSON(example);
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = fromJSON<Date>(result);
      expect(back).toBeInstanceOf(Date);
      expect(back.toISOString()).toBe(example.toISOString());
    });
  });
  describe('toJSONAsync', () => {
    it('supports Date', async () => {
      const example = new Date('2023-03-14T11:16:24.879Z');
      const result = await toJSONAsync(Promise.resolve(example));
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = await fromJSON<Promise<Date>>(result);
      expect(back).toBeInstanceOf(Date);
      expect(back.toISOString()).toBe(example.toISOString());
    });
  });
});
