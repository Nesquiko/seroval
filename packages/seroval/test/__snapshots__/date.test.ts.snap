// Vitest Snapshot v1

exports[`Date > serialize > supports Date 1`] = `"new Date(\\"2023-03-14T11:16:24.879Z\\")"`;

exports[`Date > serializeAsync > supports Date 1`] = `"Promise.resolve(new Date(\\"2023-03-14T11:16:24.879Z\\"))"`;

exports[`Date > toJSON > supports Date 1`] = `"{\\"t\\":{\\"t\\":11,\\"i\\":0,\\"s\\":\\"2023-03-14T11:16:24.879Z\\"},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`Date > toJSONAsync > supports Date 1`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":11,\\"i\\":1,\\"s\\":\\"2023-03-14T11:16:24.879Z\\"}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;
